### PR TITLE
feat: [CDS-57730]:ecs-infra-provisioner

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
Added provisioner field for ECS infra
Screenshot 2023-03-22 at 16 10 16
Screenshot 2023-03-22 at 16 10 36

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by litvinovstas** [comment]
retrigger jest

**Comment by litvinovstas** [comment]
retrigger coverage

**Comment by litvinovstas** [comment]
retrigger coverage

**Comment by litvinovstas** [comment]
retrigger cypress-pipeline

File: src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpecEditable.tsx [hunk 1]
```diff
@@ -170,6 +173,9 @@ export const ECSInfraSpecEditable: React.FC<ECSInfraSpecEditableProps> = ({
           formikRef.current = formik as FormikProps<unknown> | null
           return (
             <FormikForm>
+              <Layout.Horizontal className={css.formRow} spacing="medium">
+                <ProvisionerField name="provisioner" isReadonly />
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpecEditable.tsx:177]: `isReadonly` will always be sent as `true`, intentional?

Comment (litvinovstas) [reply src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpecEditable.tsx:177]: Yes, from this part it should be always disabled

File: src/modules/70-pipeline/components/Provisioner/__tests__/ProvisionerField.test.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      <TestWrapper>
+        <Formik initialValues={{ provisioner: 'test1' }} onSubmit={() => undefined} formName="TestWrapper">
+          <ProvisionerField name="provisioner" />
+        </Formik>
+      </TestWrapper>
+    )
+
+    const checkBox = getByText('pipeline.provisionerField.description')
+    fireEvent.click(checkBox)
+    expect(checkBox).toBeTruthy()
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/Provisioner/__tests__/ProvisionerField.test.tsx:33]: Add one more assertion that checks if text field is displayed.

File: src/modules/70-pipeline/components/Provisioner/__tests__/ProvisionerSelect.test.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            provisioners={[...provisionersMock] as ExecutionWrapperConfig[]}
+            path={''}
+          />
+        </Formik>
+      </TestWrapper>
+    )
+
+    const provSelectField = findByTestId('provisioner-select')
+
+    expect(provSelectField).toBeDefined()
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/Provisioner/__tests__/ProvisionerSelect.test.tsx:40]: use `toBeInTheDocument` here.
Also add more assertions to check if intended dropdown options are displayed.

File: src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpecEditable.tsx (hunk 1 above)
Comment (litvinovstas) [reply src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpecEditable.tsx:177]: Yes, this is requirement from PM

**Comment by litvinovstas** [comment]
retrigger coverage

**Comment by litvinovstas** [comment]
retrigger coverage

**Comment by litvinovstas** [comment]
retrigger coverage

**Review by rameshx — DISMISSED** [review]

File: src/modules/70-pipeline/components/Provisioner/__tests__/ProvisionerSelect.test.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            path={''}
+          />
+        </Formik>
+      </TestWrapper>
+    )
+    const provSelectField = container.querySelector('.bp3-input-group')
+    expect(provSelectField).toBeInTheDocument()
+
+    fireEvent.click(provSelectField!)
+    await waitFor(() => expect(container.querySelector('.bp3-menu')).toBeNull())
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/Provisioner/__tests__/ProvisionerSelect.test.tsx:41]: `bp3-menu` should be visible when you open dropdown, right ?
May be, you meant `await waitFor(() => expect(container.querySelector('.bp3-menu')).not.toBeNull())` ??

**Review by chetanvekariya-harness — APPROVED** [review]

**Comment by litvinovstas** [comment]
retrigger cypress-pipeline

**Comment by litvinovstas** [comment]
retrigger cypress-pipeline

**Comment by litvinovstas** [comment]
retrigger cypress-cv

**Review by rameshx — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Comment by litvinovstas** [comment]
retrigger cypress-cv

**Pull request merged**